crates/daphne-server/src/lib.rs

```diff
@@ -100,6 +112,38 @@ impl router::DaphneService for App {
     fn signing_key(&self) -> Option<&p256::ecdsa::SigningKey> {
         self.service_config.signing_key.as_ref()
     }
+
+    fn check_bearer_token(
+        &self,
+        token: &BearerToken,
+        sender: DapSender,
+        task_id: Option<TaskId>,
+    ) -> Result<(), String> {
+        match (&self.service_config.peer_auth, sender, task_id) {
+            (PeerAuth::Helper { leader_token }, DapSender::Leader, _) if leader_token == token => {
+                Ok(())
+            }
+            (PeerAuth::Leader { collector_token }, DapSender::Collector, _)
+                if collector_token == token =>
+            {
+                Ok(())
+            }
+            #[cfg(feature = "test-utils")]
+            (_, sender, Some(task_id)) => {
+                if self.interop_bearer_tokens.matches(sender, task_id, token) {
+                    Ok(())
+                } else {
+                    Err(format!(
+                        "the indicated bearer token is incorrect for the {sender:?} with task_id {}",
+                        daphne::messages::Base64Encode::to_base64url(&task_id),
+                    ))
+                }
+            }
```

Comment:
So the fact this code is behind "test-utils" gives me a hint: by "interop bearer token" you perhaps mean the set of bearer tokens that would have been configured via https://divergentdave.github.io/draft-dcook-ppm-dap-interop-test-design/draft-dcook-ppm-dap-interop-test-design.html. Is that right?

If so, I think you're missing an important detail. The original vision was that each task would be associated with a method or methods for authorization, e.g., the bearer token for the leader. I don't want us to lose cite of this vision. If we're changing this part of the code, I think it would be a good time to make sure what we build can be made multitenant.

Reply:
what do you mean by "each task would be associated with a method or methods for authorization, e.g., the bearer token for the leader"? The way I wrote the code each task has an associated bearer token, that's why `matches` takes a `task_id`